### PR TITLE
function of create-game finished

### PR DESCRIPTION
**Pull request by dtdoris**
A. create-game function finished
1. When the user is not logged in, and the user goes to '/create-game.html', the view page would redirect into '/login-register.html'
2. After the user sign up and login in in '/login-register.html', then the user goes to '/create-game.html' again, at this time, the view page could know the user is logged in (by adding session() in login-register.py) and could read user_id automatically. Then the view page allows the logged user to insert contents and create new wordle game.
3. If the user inputs a game name which is already existed in database, then the view page would redirect to itself refreshed after user clicks 'submit and create' button.
4. If the user inputs all valid contents and successfully create the new wordle game, then the view page would redirect into the play-game view page with the corresponding suffix. In other words, the user could directly start play this new wordle game after creating.

B. modified login-register
Including login-register.py, login-register.html, login-register.js

C. modified application.models
1. Change User as User(UserMixin, db.Model), add using UserMixin (and so we could check login state of user and get user_id easily)

D. modified __init__.py
1. Add routes
2. Initiate login manager

E. Add home page
Including home.html, home.css, home.py
It is the main art element from login-regieter view page, but instead of login windows, adding navigation bar elements.

Things need concerns:
1. I should using a modal window to show fresh message when the user fails to create a new game because the game name is not unique. Or using a new view page. Or show some color change and text words around the text box of game name. 
2. Even though now I set in login-register.py that after logged in, the view page should redirect to puzzle-list page. But first it does not redirect actually, and second I realize now that the logic should better be redirect to home page.
3. Test suites for comprehensive automatic testing are needed.

### CODE REVIEW

**Review by wuryantoAji — APPROVED** [review]
Great work, this is a much better login solution. Nice work Chuanrui

**Review by SoleilGU — CHANGES_REQUESTED** [review]
I feel that a home page is unnecessary, as indicated in our mockup. The entire logic of the webpage design is that when users open our website, they must first choose to either log in/register or play as a guest. The former two options provide access to their game records/created games and scores, which require database storage of user information. Players logging in as guests can only play games, and their scores do not get stored in the database. Following this logic, the login-register page is a necessary prerequisite for users to access the games. If a home page is added, I would like to understand its purpose. It is simply not necessary logically. In other words, this page is highly redundant.
I'm wondering why you add a home page here, please give me some convincing reasons.

**Comment by SoleilGU** [comment]
I've checked the home.html. It only removes the three buttons of login/ register/ guest and adds a navigation, but as I mentioned before, the home page doesn't provide any useful information or interaction for users. I consider this page to be redundant, and it would be best to remove it to avoid disrupting our website's original logic.

**Comment by wuryantoAji** [comment]
aaa that's a good one Yuxin. I think it's better if we stick to the mockup and make the login-register page as our "homepage". we can add a redirect in the __init__ to redirect the default route to login-register so we don't actually need a home page

**Review by wuryantoAji — CHANGES_REQUESTED** [review]
Also another thing, the login_register.html should not be changed because it should use the WTForm flask. please revert these files
1. login-register.html
2. home.py
3. home.html
4. home.css
5. login_register.js

File: application/login_register.py [hunk 1]
```diff
@@ -25,39 +25,26 @@ def login_register():
     loginForm = LoginForm()
     registerForm = RegisterForm()
     if request.method == 'POST':
-        if registerForm.hiddenTag.data == 'register':
```
Comment (wuryantoAji) [inline application/login_register.py]: revert this part

File: application/login_register.py [hunk 2]
```diff
@@ -25,39 +25,26 @@ def login_register():
...
+                return jsonify(success=False, message='Username already exists!')
+            new_user = User(username=username, email=email)
+            new_user.set_password(password)
             db.session.add(new_user)
             db.session.commit()
-            flash('Registration successful.','message')
-            return redirect(url_for('login-register.login_register'))
-        elif loginForm.hiddenTag.data == 'login':
-            # Login
-            username = loginForm.username.data
```
Comment (wuryantoAji) [inline application/login_register.py:45]: revert this one

File: application/login_register.py [hunk 3]
```diff
@@ -25,39 +25,26 @@ def login_register():
...
+            password = request.form['password']
             user = User.query.filter_by(username=username).first()
-            if user and check_password(user.password_hash, password):
-                session['user_id'] = user.user_id
-                flash('You have been successfully logged in.','message')
-                return redirect(url_for('puzzle-list.index'))
-            flash('Invalid username or password!','error')
-            return redirect(url_for('login-register.login_register')) # add flash or something
-    elif request.method == 'GET':
-        return render_template('login_register.html', loginForm=loginForm, registerForm=registerForm)
```
Comment (wuryantoAji) [inline application/login_register.py:55]: revert this one

**Comment by dtdoris** [comment]
The new files and changes are integrated into branch "leaderboard" and the mentioned modifications of the current request are discussed and solved already.

**Pull request closed without merging**